schemas/towns_fund.json

```diff
@@ -120,6 +120,15 @@
         "parent_table": "Project_Dim",
         "parent_pk": "Project_ID"
       }
+    },
+    "enums": {
+      "Status": [
```

Comment:
this is starting to feel a bit repetitious. Not sure if it's a problem, but for example, if the schema was written in python rather than json, the enum's could be more easily defined as something like "status": StatusEnum (rather than verbosely repeating all the options, then checking they match). What do you think?

Reply:
Since editing the schema will require inherent knowledge of the code anyway, even if we kept JSON, we could just switch this section to reference the enum name, then the schema validator can just check the name matches the actual Enum class?

Reply:
And the same in all the validation code. Can just check a field value is in an enum class, rather than in a recreated list?

Reply:
I think I agree to some extent. I think a consideration of your proposed approach is that we're then coupling the validation with the rest of the application code when at the moment the validation is completely separate

Reply:
>  then writing long verbose test assertions

what do you mean by this?

Reply:
We could alternatively define the enums separately in the JSON and then reference them by each sheet. i.e. define 'status' enum once and then reference it multiple times throughout the schema.

Reply:
Yeah I get your point re decoupling validation from the code. 

However, I think to some degree that this difference is academic to some point (at least at the moment) as the data itself will still ultimately have to synchronously pass both validation and then ingest into a DB. 

The idea of the validation is to capture errors that will prevent the data being inserted into our DB, and usefully represent them all to a user (rather than DB insert failure on the first incorrect instance). So if data fails validation, that data should not be able to be ingested into the DB without error. Likewise, in the case where the DB changes, and validation doesn't, ingest should also fail. So at the moment they are coupled. 

I think the dynamic / decoupled part is the idea that we may have further configurable data structures for different funds, but I think this is currently beyond our code - just a concept with no real requirements yet.

Reply:
> > then writing long verbose test assertions
> 
> what do you mean by this?

Actually, yeah they are not long, sorry (3 values is not long). But if we changed schema mechanism to use actual enum's rather than manually defined lists, we could create a test example enum, or even use a real one (from the code) for our tests, rather than writing a test that passes in a reconstructed enum equivalent like `"Non Existent Column": ["EnumValueA", "EnumValueB", "EnumValueC"]`. 

If the schema was updated to use an enum, the test could simply reference one too, which I think then also has the advantage of making it clearer what the test is testing.

Reply:
> r

but then we have 2 sources of truth for enum definition.

Reply:
using `json` provides no benefit with the current requirements and is much less useful than just using `python`